### PR TITLE
Process study ui

### PR DESCRIPTION
**Pull request by josenavas**
Contains the code from https://github.com/biocore/qiita/pull/1634/files but fixing the comments that I have.

@antgonza @mortonjt @ElDeveloper able to review?

### CODE REVIEW

File: qiita_pet/handlers/api_proxy/artifact.py [hunk 1]
```diff
@@ -17,6 +17,52 @@
...
+        return access_error
+
+    can_submit_ebi = artifact.can_be_submitted_to_ebi
+    ebi_run_accessions = (artifact.ebi_run_accessions
+                          if can_submit_ebi else None)
+    can_submit_vamps = artifact.can_be_submitted_to_vamps
+    is_submitted_vamps = (artifact.is_submitted_to_vamps
+                          if can_submit_vamps else False)
+
+    return {'id': artifact.id,
```
Comment (antgonza) [inline qiita_pet/handlers/api_proxy/artifact.py]: can this be artifact_id?

File: qiita_pet/templates/study_ajax/processing_study.html [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+      var node_id = nodes.length + 1;
+      edges.add({ id: edges.length + 1, from: command_node, to: node_id});
+      nodes.add({ id: node_id, label: command_info.output[i], group: 'type', command_id: null});
+    }
+    proc_network.redraw();
+  }
+
+  function remove_command(node_id) {
+    //TODO: delete job callback
+    if(node_id.length > 1) { alert("Can only delete one command at a time."); return;}
```
Comment (antgonza) [inline qiita_pet/templates/study_ajax/processing_study.html]: This is done more than once. Perhaps a function (function validate_node_id) that also returns an int?

**Comment by antgonza** [comment]
Looks good, some minor comments. Also, note the errors due to load_artifact and flake8.

File: qiita_pet/handlers/api_proxy/artifact.py (hunk 1 above)
Comment (josenavas) [inline qiita_pet/handlers/api_proxy/artifact.py]: Done

File: qiita_pet/templates/study_ajax/processing_study.html (hunk 2 above)
Comment (josenavas) [inline qiita_pet/templates/study_ajax/processing_study.html]: Done

**Comment by josenavas** [comment]
Thanks @antgonza

**Comment by antgonza** [comment]
You welcome. Are you gonna address the errors (load_artifact and flake8)?

**Comment by josenavas** [comment]
Good call - couldn't reproduce the flake8 errors locally but I think I fix them. The load_artifact error seems a result of a previous merge conflict, o/w I don't understand why those parameters where back in the command.

Thanks!

File: qiita_pet/test/test_study_handlers.py [hunk 3]
```diff
@@ -686,8 +686,17 @@ def test_post_admin(self):
         self.assertEqual(Artifact(3).visibility, 'sandbox')
 
 
-@qiita_test_checker()
 class TestAddArtifact(TestHandlerBase):
+    database = True
```
Comment (mortonjt) [inline qiita_pet/test/test_study_handlers.py]: Is this unused?

File: qiita_pet/templates/study_ajax/processing_study.html [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    };
+    proc_network = new vis.Network(container, data, options);
+    proc_network.on('select', function(params) {
+          //render the information for the selected artifacts or commands
+          if(params.nodes.length !== 0) {
+            // Make sure all selected nodes are of same group, so we dont link files to commands
+            var group = nodes.get(params.nodes[0]).group;
+            for(var i=0;i<params.nodes.length;i++) { if(nodes.get(params.nodes[i]).group != group) { $("#network-info").empty(); }}
+            curr_nodes = params.nodes;
+            if(group == 'command') { load_job(params.nodes); }
```
Comment (mortonjt) [inline qiita_pet/templates/study_ajax/processing_study.html]: Don't you want === instead?

Comment (mortonjt) [inline qiita_pet/templates/study_ajax/processing_study.html]: nvm

File: qiita_pet/templates/study_ajax/processing_study.html [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+  //currently selected nodes
+  var curr_nodes = [1];
+
+  function validate_node_id(node_id, action){
+      if (node_id.length !== 1) {
+          alert("Can only " + action + " one command at a time");
+          return -1;
+      }
+      // node_id only has one element
+      return +node_id[0];
```
Comment (mortonjt) [inline qiita_pet/templates/study_ajax/processing_study.html]: Never seen this +operation used in this context before.  What does it do out of curiosity?

**Comment by mortonjt** [comment]
No immediate problems that I see

File: qiita_pet/templates/study_ajax/processing_study.html (hunk 5 above)
Comment (josenavas) [inline qiita_pet/templates/study_ajax/processing_study.html]: That is used for casting to int. IMO, I don't think it is needed here, but I haven't played enough with this code to be 100% sure, so I prefer to leave as it is.

File: qiita_pet/test/test_study_handlers.py (hunk 3 above)
Comment (josenavas) [inline qiita_pet/test/test_study_handlers.py]: It looks weird, but `database` is a class attribute from `TestHandlerBase` and it is used in the `setUp` from that class to decide whether we need to reset the DB or not after each test.

**Comment by antgonza** [comment]
Merging as @josenavas has answered to @mortonjt questions.

**Pull request merged**